### ISSUE TITLE
different behavior in signing process. resulting error of signature output

### ISSUE THREAD

**Issue by djkampus** [trufflesuite/ganache-ui#5208]
##tl;dr
in concatenated signature, trufflesuite produce wrong signature return in last byte.

## Expected Behavior

console can return proper and valid signature of messageHash. 

## Current Behavior

signing process silently success,but the result are just one string of concatenated string.

i call the function like this
`web3.eth.sign(messageHash,accounts[0])`
and my only return just one long concatenated string
'0x6ac2fc574b9f7cb29133c73a3dbc68e00d705454cc2262aa507ed7a68d716c141b7f52c98cc2a93a783a89e8da60985734c5f6e96712fce702118be586ff7ecb01'

while i thoroughly examine the result i found that my signature that comes from truffle suite and my remix IDE console are totally different.

from remixIDE i got this with similar web3.eth.sign() process
`{"message":"0xb4b2299e9e98ac08378f2afe853c9eb0979e1547bc150e8a6206cfed13ad4b6c","messageHash":"0x5d53969665f4e9fb4622f36525a3787037be16c9f57dd070e41e1080abfeead9","v":"0x1c","r":"0x6ac2fc574b9f7cb29133c73a3dbc68e00d705454cc2262aa507ed7a68d716c14","s":"0x1b7f52c98cc2a93a783a89e8da60985734c5f6e96712fce702118be586ff7ecb","signature":"0x6ac2fc574b9f7cb29133c73a3dbc68e00d705454cc2262aa507ed7a68d716c141b7f52c98cc2a93a783a89e8da60985734c5f6e96712fce702118be586ff7ecb1c"} `

in concatenated signature, trufflesuite produce wrong signature return in last byte.

## Possible Solution
maybe just simply return web3.eth.sign() with full information of signature like this will work best(like remixIDE console).
``{"message":"0xb4b2299e9e98ac08378f2afe853c9eb0979e1547bc150e8a6206cfed13ad4b6c","messageHash":"0x5d53969665f4e9fb4622f36525a3787037be16c9f57dd070e41e1080abfeead9","v":"0x1c","r":"0x6ac2fc574b9f7cb29133c73a3dbc68e00d705454cc2262aa507ed7a68d716c14","s":"0x1b7f52c98cc2a93a783a89e8da60985734c5f6e96712fce702118be586ff7ecb","signature":"0x6ac2fc574b9f7cb29133c73a3dbc68e00d705454cc2262aa507ed7a68d716c141b7f52c98cc2a93a783a89e8da60985734c5f6e96712fce702118be586ff7ecb1c"} ``
so, when needed we just take the part we want without hassle because we already have all of them. 
no need to split r,s,v;or checking the message and messageHash , and we already have the full signature 

## Steps to Reproduce (for bugs)

1.i have already prepared the messageHash to be signed.
2.invoke web3.eth.sign() from truffle console.
3.when i trying to pass it as argument, Openzeppelin ECDSA revert error with invalid signature
4.i invoked web3.eth.sign() from remix console and pass full signature manually. then the library returns the proper result.

## Context

it affects my workflow because signature output need to check chars-by-chars. when something wrong, it not only takes time to examine, maybe it will compromise security solutions

## Your Environment

* Link to your project:(sorry its a private work)
* Truffle v5.7.1 (core: 5.7.1)
Ganache v7.6.0
Solidity - 0.8.17 (solc-js)
Node v18.12.1
Web3.js v1.8.1

**Comment by djkampus** [comment]
maybe i forgot
i invoke the web3.eth.sign while im already inside truffle console

**Comment by djkampus** [comment]
ah yes
im using VSCode version of trufflesuite

**Comment by davidmurdoch** [comment]
Thanks for opening this issue. It looks like we are not adding EIP-155 replay protection to the signature. The last byte(s) should be computed with the chainId. Some more investigation is needed to determine the proper fix.

**Comment by djkampus** [comment]
i will try again with my trouble here.

i compare the signature verification process here 
``https://etherscan.io/verifySig/13420``

and i have my function as below
`      function checkSigRaw(bytes32 _messageHash,bytes memory _signature,address _address)public pure returns(bool){
        address output = ECDSA.recover(_messageHash, _signature);
        return (_address == output);
      }`

from etherscan verify, signature verified.
but  while check it with my function above. ECDSA revert as error.

**Comment by djkampus** [comment]
> ##tl;dr in concatenated signature, trufflesuite produce wrong signature return in last byte.
> ## Expected Behavior
> 
> console can return proper and valid signature of messageHash.
> ## Current Behavior
> 
> signing process silently success,but the result are just one string of concatenated string.
> 
> i call the function like this `web3.eth.sign(messageHash,accounts[0])` and my only return just one long concatenated string '0x6ac2fc574b9f7cb29133c73a3dbc68e00d705454cc2262aa507ed7a68d716c141b7f52c98cc2a93a783a89e8da60985734c5f6e96712fce702118be586ff7ecb01'
> 
> while i thoroughly examine the result i found that my signature that comes from truffle suite and my remix IDE console are totally different.
> 
> from remixIDE i got this with similar web3.eth.sign() process `{"message":"0xb4b2299e9e98ac08378f2afe853c9eb0979e1547bc150e8a6206cfed13ad4b6c","messageHash":"0x5d53969665f4e9fb4622f36525a3787037be16c9f57dd070e41e1080abfeead9","v":"0x1c","r":"0x6ac2fc574b9f7cb29133c73a3dbc68e00d705454cc2262aa507ed7a68d716c14","s":"0x1b7f52c98cc2a93a783a89e8da60985734c5f6e96712fce702118be586ff7ecb","signature":"0x6ac2fc574b9f7cb29133c73a3dbc68e00d705454cc2262aa507ed7a68d716c141b7f52c98cc2a93a783a89e8da60985734c5f6e96712fce702118be586ff7ecb1c"} `
> 
> in concatenated signature, trufflesuite produce wrong signature return in last byte.
> ## Possible Solution
> 
> maybe just simply return web3.eth.sign() with full information of signature like this will work best(like remixIDE console). `{"message":"0xb4b2299e9e98ac08378f2afe853c9eb0979e1547bc150e8a6206cfed13ad4b6c","messageHash":"0x5d53969665f4e9fb4622f36525a3787037be16c9f57dd070e41e1080abfeead9","v":"0x1c","r":"0x6ac2fc574b9f7cb29133c73a3dbc68e00d705454cc2262aa507ed7a68d716c14","s":"0x1b7f52c98cc2a93a783a89e8da60985734c5f6e96712fce702118be586ff7ecb","signature":"0x6ac2fc574b9f7cb29133c73a3dbc68e00d705454cc2262aa507ed7a68d716c141b7f52c98cc2a93a783a89e8da60985734c5f6e96712fce702118be586ff7ecb1c"} ` so, when needed we just take the part we want without hassle because we already have all of them. no need to split r,s,v;or checking the message and messageHash , and we already have the full signature
> ## Steps to Reproduce (for bugs)
> 
> 1.i have already prepared the messageHash to be signed. 2.invoke web3.eth.sign() from truffle console. 3.when i trying to pass it as argument, Openzeppelin ECDSA revert error with invalid signature 4.i invoked web3.eth.sign() from remix console and pass full signature manually. then the library returns the proper result.
> ## Context
> 
> it affects my workflow because signature output need to check chars-by-chars. when something wrong, it not only takes time to examine, maybe it will compromise security solutions
> ## Your Environment
> 
>     * Link to your project:(sorry its a private work)
> 
>     * Truffle v5.7.1 (core: 5.7.1)
>       Ganache v7.6.0
>       Solidity - 0.8.17 (solc-js)
>       Node v18.12.1
>       Web3.js v1.8.1

i've just test again with etherscan verify this signature. and it work
but still return to false in trufflesuite

`https://etherscan.io/verifySig/13421`

**Comment by davidmurdoch** [comment]
I haven't fixed the bug yet.
